lib/LANraragi/Plugin/Metadata/Chaika.pm

```diff
@@ -43,17 +47,17 @@ sub get_tags {
     # Parse the given link to see if we can extract type and ID
     my $oneshotarg = $lrr_info->{oneshot_param};
     if ( $oneshotarg =~ /https?:\/\/panda\.chaika\.moe\/(gallery|archive)\/([0-9]*)\/?.*/ ) {
-        ( $newtags, $newtitle ) = tags_from_chaika_id( $1, $2 );
+        ( $newtags, $newtitle ) = tags_from_chaika_id( $1, $2, $addother, $addsource );
     } else {
 
         # Try SHA-1 reverse search first
         $logger->info("Using thumbnail hash " . $lrr_info->{thumbnail_hash});
-        ( $newtags, $newtitle ) = tags_from_sha1( $lrr_info->{thumbnail_hash} );
+        ( $newtags, $newtitle ) = tags_from_sha1( $lrr_info->{thumbnail_hash}, $addother, $addsource );
 
         # Try text search if it fails
         if ( $newtags eq "" ) {
             $logger->info("No results, falling back to text search.");
-            ( $newtags, $newtitle ) = search_for_archive( $lrr_info->{archive_title}, $lrr_info->{existing_tags} );
+            ( $newtags, $newtitle ) = search_for_archive( $lrr_info->{archive_title}, $lrr_info->{existing_tags}, $addother, $addsource );
         }
     }
 
```

Comment:
Instead of carrying the `$addsource` variable across all subs, you could just check for it here:  

```
if ($addsource ne "") {
    $newtags .= ", source:" . $addsource;
}
```
(I think the plugin system is lenient enough to allow the comma at the start even if no other tags came in, but might be good to check)

Reply:
That's a good catch, thanks. I will put it after the no tags check on line 65 so adding strings like that should be fine.

Reply:
Actually, it was awkward to test so I decided to carry $addsource across the subs after all.